src/material/checkbox/checkbox.spec.ts

```diff
@@ -466,6 +466,19 @@ describe('MDC-based MatCheckbox', () => {
       expect(inputElement.disabled).toBe(false);
     }));
 
+    it('should not change the checked state if disabled and interactive', fakeAsync(() => {
```

Comment:
This is unrelated to the radio button, but I realized that I forgot to include it in #29474.